driver-core/src/test/unit/com/mongodb/internal/connection/AuthorizationHeaderTest.java

```diff
@@ -0,0 +1,174 @@
+
+import static org.junit.Assert.assertEquals;
+
+public class AuthorizationHeaderTest {
+    private final String timestamp = "20150830T123600Z";
+
+    @Test
+    public void testHash() throws SaslException {
+        String actual = AuthorizationHeader.hash("");
+        String expected = "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855";
+        assertEquals(expected, actual);
+
+        String request = "GET\n"
+                + "/\n"
+                + "Action=ListUsers&Version=2010-05-08\n"
+                + "content-type:application/x-www-form-urlencoded; charset=utf-8\n"
+                + "host:iam.amazonaws.com\n"
+                + String.format("x-amz-date:%s\n", timestamp)
+                + "\n"
+                + "content-type;host;x-amz-date\n"
+                + "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855";
+
+        actual = AuthorizationHeader.hash(request);
+        expected = "f536975d06c0309214f805bb90ccff089219ecd68b2577efef23edd43b7e1a59";
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testGetRegion() throws SaslException {
+        String actual = AuthorizationHeader.getRegion("sts.amazonaws.com");
+        String expected = "us-east-1";
+        assertEquals(expected, actual);
+
+        actual = AuthorizationHeader.getRegion("first");
+        assertEquals(expected, actual);
+
+        actual = AuthorizationHeader.getRegion("first.second");
+        expected = "second";
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testGetCanonicalHeaders() {
+        Map<String, String> headers = new HashMap<>();
+
+        headers.put("Host", "iam.amazonaws.com");
+        headers.put("Content-Type", "application/x-www-form-urlencoded; charset=utf-8");
+        headers.put("My-header1", "    a   b   c  ");
+        headers.put("X-Amz-Date", timestamp);
+        headers.put("My-Header2", "    \"a   b   c\"  ");
+
+        String actual = AuthorizationHeader.getCanonicalHeaders(headers);
+        String expected = "content-type:application/x-www-form-urlencoded; charset=utf-8\n"
+                + "host:iam.amazonaws.com\n"
+                + "my-header1:a b c\n"
+                + "my-header2:\"a b c\"\n"
+                + String.format("x-amz-date:%s\n", timestamp);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testGetSignedHeaders() {
+        Map<String, String> headers = new HashMap<>();
+
+        headers.put("Host", "iam.amazonaws.com");
+        headers.put("Content-Type", "application/x-www-form-urlencoded");
+        headers.put("X-Amz-Date", timestamp);
+
+        String actual = AuthorizationHeader.getSignedHeaders(headers);
+        String expected = "content-type;host;x-amz-date";
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testCreateCanonicalRequest() throws SaslException {
+        Map<String, String> parameters = new HashMap<>();
```

Comment:
This map is not used in the test.  Is it supposed to be?

Reply:
Ah, this is no longer used. Removed.